airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -144,6 +145,25 @@ def ti_run(
         if not dr:
             raise ValueError(f"DagRun with dag_id={dag_id} and run_id={run_id} not found.")
 
+        # Clear XCom data for the task instance since we are certain it is executing
+        # However, do not clear it for deferral
+        task_instance = select(TI.dag_id, TI.task_id, TI.run_id, TI.map_index, TI.next_method).where(
+            TI.id == ti_id_str
+        )
```

Comment:
Should we get this too at L77 instead? wdyt?

Reply:
Yeah we can, since we handle only "one" situation in "run", we can do that

Reply:
Handled in latest commit